components/uniset/src/uniset.rs

```diff
@@ -171,11 +191,11 @@ impl UnicodeSet {
     ///
     /// `let condition_closure = |_: usize| -> bool {true};`
     /// `self.contains(10, condition_closure);`
-    fn contains<C>(&self, query: u32, condition: C) -> bool
+    fn contains<C>(&self, query: char, condition: C) -> bool
```

Comment:
Idea: make `.contains_impl` return an `Option<Range<u32>>`, and then you can implement both `.contains` and `.contains_range` without the unintuitive closure trick.

Reply:
agreed and changed